test/e2e/framework/config_test.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework_test
+
+import (
+	"testing"
+
+	"github.com/onsi/gomega"
+
+	frameworkx "sigs.k8s.io/cluster-api-provider-vsphere/test/e2e/framework"
+)
+
+func TestLoadConfig(t *testing.T) {
+	g := gomega.NewWithT(t)
+	config, err := frameworkx.LoadConfig([]byte(exampleConfigYAML1))
+	g.Expect(err).ShouldNot(gomega.HaveOccurred())
+	g.Expect(config).ShouldNot(gomega.BeNil())
+
+	g.Expect(config.Images).To(gomega.HaveLen(1))
+	g.Expect(config.Images[0]).To(gomega.Equal("capv-manager:e2e"))
+
+	g.Expect(config.Components).To(gomega.HaveLen(5))
+
+	g.Expect(config.Components[0].Waiters).To(gomega.HaveLen(2))
+	g.Expect(config.Components[0].Waiters[1].Type).To(gomega.Equal(frameworkx.PodsWaiter))
+}
+
+const exampleConfigYAML1 = `---
+images:
+- capv-manager:e2e
+
+components:
+
+# Load the certificate manager and wait for all of its pods and service to
+# become available.
+- name:    cert-manager
+  sources:
+  - type:  url
+    value: https://github.com/jetstack/cert-manager/releases/download/v0.11.1/cert-manager.yaml
+  waiters:
+  - type:  service
+    value: v1beta1.webhook.cert-manager.io
+  - value: cert-manager
+
+# Load CAPI core and wait for its pods to become available.
+- name:    capi
+  sources:
+  - value: https://github.com/kubernetes-sigs/cluster-api//config/default?ref=master
+  waiters:
+  - value: capi-system
+
+# Load the CAPI kubeadm bootstrapper and wait for its pods to become available.
+- name:    capi-kubeadm-bootstrap
+  sources:
+  - value: https://github.com/kubernetes-sigs/cluster-api//bootstrap/kubeadm/config/default?ref=master
+  waiters:
+  - value: capi-kubeadm-bootstrap-system
+
+# Load the CAPI kubeadm control plane and wait for its pods to become available.
+- name:    capi-kubeadm-control-plane
+  sources:
+  - value: https://github.com/kubernetes-sigs/cluster-api//controlplane/kubeadm/config/default?ref=master
+  waiters:
+  - value: capi-kubeadm-control-plane-system
+
+# Load CAPV and wait for its pods to become available.
+- name:    capv
+  sources:
+  - value: ../../config/default
+    replacements:
+    - old: gcr.io/cluster-api-provider-vsphere/release/manager:latest
+      new: capv-manager:e2e
+  waiters:
+  - value: capv-system
```

Comment:
Does this mean we are waiting for the whole pods of the namespace to be ready ? Is value accepting pods or namespace values ?

Reply:
I'll document this.

Reply:
Hi @yastij,

I don't think it's necessary to document these values every place they're used. I'm documenting them in the `ComponentWaiterTypes` in source. The documentation for the component also indicates that the logic is blocked contingent upon the pods being available.